### PR TITLE
chore: update loadtesting to be more reflective of real world app

### PR DESCRIPTION
**Pull request by markphelps**
redoes our loadtest to do the following:

- vary the request payload for each call, allowing us to reflect real world usage better
- enable caching and auth separately to see how they affect performance
- change the rate of requests
- set log levels on both Flipt and Pyroscope to > DEBUG to remove extraneous output in the log
- run both variant eval (v1) and boolean eval (rollouts) separately for 60s each, outputting results at end

## Example:

```
$ FLIPT_AUTH_ENABLED="true" mage hack:loadTest

[EVALUATE]: Vegeta attack for 60.000000 seconds...
[EVALUATE]: Vegeta attack complete

Authentication Enabled: true
Cache Enabled: false
--------------------
Mean: 12.443488ms
95th percentile: 13.901631ms
99th percentile: 17.551379ms
Max: 45.307333ms
Requests: 6000
Throughput: 100.012008
Errors: []
Status Codes: map[200:6000]
Success: 100.000000 %


[BOOLEAN]: Vegeta attack for 60.000000 seconds...
[BOOLEAN]: Vegeta attack complete

Authentication Enabled: true
Cache Enabled: false
--------------------
Mean: 4.531645ms
95th percentile: 9.427763ms
99th percentile: 15.554324ms
Max: 32.3955ms
Requests: 6000
Throughput: 100.005388
Errors: []
Status Codes: map[200:6000]
Success: 100.000000 %
```

### CODE REVIEW

File: build/testing/test.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func WithSQLite(ctx context.Context, client *dagger.Client, container *dagger.Co
 
 func WithPostgres(ctx context.Context, client *dagger.Client, flipt *dagger.Container) (context.Context, *dagger.Client, *dagger.Container) {
 	return ctx, client, flipt.
-		WithEnvVariable("FLIPT_TEST_DB_URL", "postgres://postgres:password@postgres:5432").
+		WithEnvVariable("FLIPT_TEST_DB_URL", "postgres://postgres:password@postgres:5432?sslmode=disable").
```
Comment (markphelps) [inline build/testing/test.go:82]: im honestly not sure how this works without disabling SSL for unit tests with PG. It seems we disable SSL for cockroachdb already for our unit tests so I would think we would need to do the same for PG?

File: build/hack/cmd/loadtest/main.go [hunk 2]
```diff
@@ -28,80 +27,99 @@ type BooleanRollout struct {
...
-	evaluationTargets = append(evaluationTargets, vegeta.Target{
-		Method: postMethod,
-		URL:    fmt.Sprintf("%s/api/v1/evaluate", fliptAddr),
-		Body:   ve,
+	evaluationTargeter := vegeta.Targeter(func(t *vegeta.Target) error {
+		t.Header = http.Header{
+			"Authorization": []string{fmt.Sprintf("Bearer %s", fliptAuthToken)},
+		}
+		t.Method = postMethod
+		t.URL = fmt.Sprintf("%s/api/v1/evaluate", fliptAddr)
```
Comment (yquansah) [inline build/hack/cmd/loadtest/main.go:53]: Are you using the legacy endpoint on purpose right here?

**Review by yquansah — COMMENTED** [review]
This looks good to me. We can look into getting an `import.yaml` for this load test that will include segments for boolean evaluations to get a more truer picture of latency concerns

File: build/testing/test.go (hunk 1 above)
Comment (GeorgeMac) [reply build/testing/test.go:82]: Does our test suite not do this for us. Or perhaps it uses one of the modes by default that lets the server tell the client what to do (`allow`, `prefer` etc.)

**Review by GeorgeMac — APPROVED** [review]

File: build/hack/cmd/loadtest/main.go (hunk 2 above)
Comment (markphelps) [reply build/hack/cmd/loadtest/main.go]: updated to use the new new api

**Pull request merged**